cpp/include/cudf/aggregation.hpp

```diff
@@ -56,6 +56,12 @@ std::unique_ptr<aggregation> make_mean_aggregation();
 /// Factory to create a MEDIAN aggregation
 std::unique_ptr<aggregation> make_median_aggregation();
 
+/// Factory to create a NUMBA UDF aggregation
+std::unique_ptr<aggregation> make_numba_udf_aggregation();
```

Comment:
There needs to be some way to pass in the string for the UDF to these factories. Furthermore, I think we can unify them like this:
```
enum class udf_type : bool { PTX, CUDA };
std::unique_ptr<aggregation> make_udf_aggregation( udf_type type, std::string const& source);
```

Reply:
But what would be corresponding aggregation::Kind for this, should I add something like CUSTOM to `enum Kind`?

Reply:
Your addition of `NUMBA_UDF` and `CUDA_UDF` is good. You can just select between those two depending on the value of the `udf_type`.